src/map/skill.cpp

```diff
@@ -488,10 +487,8 @@ bool skill_pos_maxcount_check(struct block_list *src, int16 x, int16 y, uint16 s
 		return false;
 	}
 	if (type&battle_config.land_skill_limit && (maxcount = skill_get_maxcount(skill_id, skill_lv)) > 0) {
-		for (int i = 0; i < MAX_SKILLUNITGROUP && ud->skillunit[i] && maxcount; i++) {
-			if (ud->skillunit[i]->skill_id == skill_id)
-				maxcount--;
-		}
+		unit_skillunit_maxcount(sd->ud, skill_id, maxcount);
```

Comment:
```suggestion
		unit_skillunit_maxcount(*ud, skill_id, maxcount);
```
Needs to be the unit, because it is not always a player.